src/Features/Core/Portable/MetadataAsSource/MetadataAsSourceOptions.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.CodeAnalysis.MetadataAsSource
+{
+    /// <summary>
+    /// Options for metadata as source navigation
+    /// </summary>
+    /// <param name="NavigateToDecompiledSources"><see langword="false"/> to disallow decompiling code, which may
+    /// result in signagures only being returned if there is no other non-decompilation option available</param>
+    /// <param name="AlwaysUseDefaultSymbolServers">Whether navigation should try to use the default Microsoft and
+    /// Nuget symbol servers regardless of debugger settings</param>
+    internal readonly record struct MetadataAsSourceOptions(
+        bool NavigateToDecompiledSources = true,
+        bool AlwaysUseDefaultSymbolServers = true)
+    {
+        // note: must pass at least one parameter to avoid calling default ctor:
+        public static readonly MetadataAsSourceOptions Default = new(NavigateToDecompiledSources: true);
```

Comment:
>public static readonly MetadataAsSourceOptions Default = new(NavigateToDecompiledSources: true); [](http://example.com/codeflow?start=8&length=96)

I started using the following pattern to avoid pitfall with calling ctor w/o parameters:

```C#

public MetadataAsSourceOptions()
 : this(NavigateToDecompiledSources: true)
{
} 

public static readonly MetadataAsSourceOptions Default = new();
```

Reply:
Yeah, that's nicer. The empty constructor felt like pretty big trap